File: wek2-2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Prototipos de funciones
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{

    string text = get_string("Text: ");

    // count
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculate the values
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int count_letters(string text)
{
    int count = 0;
    int len = strlen(text);

    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i])) // Solo cuenta letras
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1; // shoud be at least one letter
    int len = strlen(text);

    for (int i = 0; i < len; i++)
    {
        if (text[i] == ' ') // a spaces indicate one word
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    int len = strlen(text);

    for (int i = 0; i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py

import cs50


def count_letters(text):
    count = 0
    for char in text:
        # if is character valid
        if char.isalpha():
            count += 1
    return count


def count_words(text):
    count = 1
    for char in text:
        if char == ' ':
            count += 1
    return count


def count_sentences(text):
    count = 0
    for char in text:
        if char in ['.', '!', '?']:
            count += 1
    return count


text = cs50.get_string("Text: ")
# count
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

# calculate the values
L = (float(letters) / words) * 100
S = (float(sentences) / words) * 100
index = 0.0588 * L - 0.296 * S - 15.8
if (index < 1):
    print("Before Grade 1")
elif (index >= 16):
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")
